classes/player/player.gd

```diff
@@ -1476,6 +1476,7 @@ func resist(val, sub, div): # ripped from source
 	return val * FPS_MOD
 
 
+@warning_ignore("shadowed_variable")
 func set_rotation_origin(facing_direction: int, origin: Vector2):
```

Comment:
We could avoid shadowing the variable by changing "facing_direction" to "direction" and multiplying origin.x by it.
i.e.
```suggestion
func set_rotation_origin(direction: int, origin: Vector2):
	# Flip origin, if appropriate.
	origin.x *= direction
	sprite.offset = origin
	sprite.position = -origin
```

Reply:
Hmm. Just `direction` is objectively more ambiguous, though.... Maybe `facing_dir`?

Good thought about multiplying `origin.x` though. Definitely a more elegant way to do it!